server/src/main/java/org/opensearch/action/admin/indices/cache/clear/TransportClearIndicesCacheAction.java

```diff
@@ -129,12 +139,31 @@ protected EmptyResult shardOperation(ClearIndicesCacheRequest request, ShardRout
             shardRouting.shardId(),
             request.queryCache(),
             request.fieldDataCache(),
-            request.requestCache(),
+            false,
             request.fields()
         );
+        // For request cache clearing, only accumulate CacheEntity for each shard, and save marking and clearing for the nodeOperation.
+        // Then the scheduled cache clear won't happen midway through.
+        ShardId shardId = shardRouting.shardId();
+        IndexShard shard = indicesService.indexService(shardId.getIndex()).getShardOrNull(shardId.id());
+
+        if ((request.requestCache() || ((request.queryCache() || request.fieldDataCache()) == false && request.fields().length == 0))
+            && shard != null) {
+            requestCacheEntities.add(new IndicesService.IndexShardCacheEntity(shard));
+        }
         return EmptyResult.INSTANCE;
     }
 
+    @Override
+    protected void nodeOperation(List<EmptyResult> results, List<BroadcastShardOperationFailedException> accumulatedExceptions) {
+        try {
+            indicesService.clearRequestCacheEntities(requestCacheEntities);
+        } catch (Exception e) {
+            clearActionLogger.warn("Node-wide force cache clear failed", e);
+        }
+        this.requestCacheEntities.clear();
```

Comment:
Considering this is a singleton object shared across different requests, there can be issues calling `clear()` when there were in-flight requests being added in `shardOperation()`. So those might be cleared out erroneously.

Only way I see to handle this would be to have a `Map<ClearIndicesCacheRequest, List<ShardId>> map;` as ClearIndicesCacheRequest is unique per request. 
Then in `shardOperation()`, you would do `map.get(request).add(shardId)`. And then in `nodeOperation()`, call `indicesService.clearCacheForShards(map.get(request))` and then clear map for this request